google/cloud/storage/emulator/gcs/holder.py

```diff
@@ -59,21 +59,31 @@ def init_upload(
 
     @classmethod
     def init_resumable_rest(cls, request, bucket):
-        name = request.args.get("name", "")
+        name = request.args.get("name", None)
         rest_only = {}
-        if len(request.data) > 0 and request.data.decode("utf-8") != "{}":
-            if name != "":
-                utils.error.invalid("name argument in non-empty payload", None)
+        if len(request.data) > 0:
             data = json.loads(request.data)
             rest_only = cls.__extract_rest_only(data)
             metadata = json_format.ParseDict(data, resources_pb2.Object())
+            if name != None and name != metadata.name:
```

Comment:
Can you add a comment to explain why the logic here is needed?

Also `name is not None` in python usually. :)